sleap/gui/web.py

```diff
@@ -146,6 +147,36 @@ def get_release(self, version: str) -> Release:
         )
 
 
+class AnnouncementChecker:
+    def __init__(self):
+        self.prefs = {"last seen announcement": None}
+
+    def check_for_new_announcement(self):
+        """Check if there is a new announcement since the last seen announcement."""
+        last_seen = self.prefs["last seen announcement"]
+        current_date = datetime.datetime.now().strftime("%Y%m%d")
+
+        if last_seen is None or current_date > last_seen:
+            self.fetch_announcement()
+
+    def fetch_announcement(self):
+        """Fetch the announcement and update the last seen date."""
+        announcement_url = "https://github.com/talmolab/sleap/releases"
+
+        try:
+            response = requests.get(announcement_url)
+            if response.status_code == 200:
+                announcement = response.text
```

Comment:
Lets expose this as an attribute to the `AnnouncementChecker` and run a test to see if we get the text we want.